packages/investor/src/InvestorOpportunity.ts

```diff
@@ -0,0 +1,100 @@
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { BigNumber } from 'bignumber.js'
+
+import { FeePriority } from './Extensions'
+
+export type DepositWithdrawArgs = { address: string; amount: BigNumber }
+
+export abstract class InvestorOpportunity<TxType = unknown, MetaData = unknown> {
+  /**
+   * Opportunity id e.g., contract address or validator address
+   */
+  readonly id: string
+  readonly displayName: string
+  readonly underlyingAsset: {
+    /**
+     * Asset to be deposited
+     *
+     * User needs an available balance of this asset to be able to deposit
+     */
+    assetId: string
+  }
+  readonly positionAsset: {
+    /**
+     * Asset that represents their position
+     */
+    assetId: string
+    /**
+     * The amount of the position asset, not the underlying asset
+     */
+    balance: BigNumber // This is probably a wallet concern not a opportunity concern
+    /**
+     * The ratio of value between the underlying asset and the position asset
+     *
+     * Multiply the position asset by this value to calculate the amount of underlying asset
+     * that will be received for a withdrawal
+     *
+     * Amount is an integer value without precision applied
+     */
+    price: BigNumber
```

Comment:
Rename the property and reword the description

Reply:
`pricePerUnderlying`, `pricePerUnderlyingAsset` or `pricePerShare` is the best I can think of. I think I prefer `pricePerUnderlying` of those three.